datacube/drivers/postgis/_api.py

```diff
@@ -673,7 +675,7 @@ def bulk_simple_dataset_search(self, products=None, batch_size: int = 0):
             )
         else:
             conn = self._connection
-        return conn.execute(query)
+        return conn.execute(query).fetchall()
```

Comment:
This and  the None-check in line 832 are the functional changes.

Reply:
Don't think we want to do this - `fetchall()` will pull the entire query in a single batch, bypassing the configured batch size.